aws/resource_aws_sagemaker_lifecycle_configuration_test.go

```diff
@@ -0,0 +1,211 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/terraform"
+
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+const SagemakerLifecycleConfigurationResourcePrefix = "tf-acc-test"
+
+func init() {
+	resource.AddTestSweepers("aws_sagemaker_lifecycle_config", &resource.Sweeper{
+		Name: "aws_sagemaker_lifecycle_config",
+		F:    testSweepSagemakerLifecycleConfiguration,
+	})
+}
+
+func testSweepSagemakerLifecycleConfiguration(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).sagemakerconn
+
+	input := &sagemaker.ListNotebookInstanceLifecycleConfigsInput{}
+	err = conn.ListNotebookInstanceLifecycleConfigsPages(input, func(page *sagemaker.ListNotebookInstanceLifecycleConfigsOutput, lastPage bool) bool {
+		if len(page.NotebookInstanceLifecycleConfigs) == 0 {
+			log.Printf("[INFO] No SageMaker notebook instance lifecycle configuration to sweep")
+			return false
+		}
+		for _, lifecycleConfig := range page.NotebookInstanceLifecycleConfigs {
+			name := aws.StringValue(lifecycleConfig.NotebookInstanceLifecycleConfigName)
+			if !strings.HasPrefix(name, SagemakerLifecycleConfigurationResourcePrefix) {
+				log.Printf("[INFO] Skipping SageMaker notebook instance lifecycle configuration: %s", name)
+				continue
+			}
+
+			log.Printf("[INFO] Deleting SageMaker notebook instance lifecycle configuration: %s", name)
+			_, err := conn.DeleteNotebookInstanceLifecycleConfig(&sagemaker.DeleteNotebookInstanceLifecycleConfigInput{
+				NotebookInstanceLifecycleConfigName: aws.String(name),
+			})
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete SageMaker notebook instance lifecycle configuration %s: %s", name, err)
+			}
+		}
+		return !lastPage
+	})
+	if err != nil {
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping SageMaker notebook instance lifecycle configuration sweep for %s: %s", region, err)
+			return nil
+		}
+		return fmt.Errorf("error retrieving SageMaker notebook instance lifecycle configuration: %s", err)
+	}
+
+	return nil
+}
+
+func TestAccAWSSagemakerLifecycleConfiguration_Basic(t *testing.T) {
+	var lifecycleConfig sagemaker.DescribeNotebookInstanceLifecycleConfigOutput
+	rName := acctest.RandomWithPrefix(SagemakerLifecycleConfigurationResourcePrefix)
+	resourceName := "aws_sagemaker_lifecycle_config.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSagemakerLifecycleConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerLifecycleConfigurationConfig_Basic(rName),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
We should check the value of the `arn` attribute is populated correct in here, e.g.

```go
testAccCheckResourceAttrRegionalARN(resourceName, "arn", "sagemaker", fmt.Sprintf("notebook-instance-lifecycle-config/%s", rName)),
```